### ISSUE TITLE
[next] Type error with CompiledMessage

### ISSUE THREAD

**Issue by banyan** [lingui/js-lingui#686]
**Describe the bug**

I'm trying this in the next version. `tsc` says the type is wrong but do I need to correct anything in my codes? 

```tsx
<Trans id="foo">hi {name}!</Trans>
```

```po
#: src/App.tsx:13
msgid "foo"
msgstr "こんにちは {name} さん"
```

```
i18n.load("ja", catalogJa.messages); // <= get incompatibility error, but works runtime if use `@ts-ignore`. 
i18n.activate("ja");
```

```
TypeScript error in /Users/banyan/tmp/lingui-cra-typescript-example/src/App.tsx(8,17):
Argument of type '{ "foo": (string | string[])[]; }' is not assignable to parameter of type 'Record<string, CompiledMessage>'.
  Property '"foo"' is incompatible with index signature.
    Type '(string | string[])[]' is not assignable to type 'CompiledMessage'.
      Type '(string | string[])[]' is not assignable to type '(string | [string, (string | undefined)?, (string | Object | undefined)?])[]'.
        Type 'string | string[]' is not assignable to type 'string | [string, (string | undefined)?, (string | Object | undefined)?]'.
          Type 'string[]' is not assignable to type 'string | [string, (string | undefined)?, (string | Object | undefined)?]'.
            Property '0' is missing in type 'string[]' but required in type '[string, (string | undefined)?, (string | Object | undefined)?]'.  TS2345
```

**To Reproduce**

```
git clone https://github.com/banyan/lingui-cra-typescript-example.git
cd lingui-cra-typescript-example
yarn
yarn lingui extract && yarn lingui compile
yarn start
```

**Expected behavior**
No compile error. 

- jsLingui version ``lingui --version``

```
❯ lingui --version
3.0.0-11
```

- Your Babel config 
Create React App

**Comment by tricoder42** [comment]
Could you please try latest pre-release (3.0.0-13)? If it still doesn't work, could you please post full snippet? I.e. from where the `catalogJa` variable comes from?

**Comment by Minishlink** [comment]
Reproducing this on 3.0.0-13.

```
js

<WelcomeTitle>
   {i18n._(/*I18n: Welcome text */ t`Welcome to ${title}`)}
</WelcomeTitle>
```

Got the typescript error but worse is : on runtime in development, it works great; but in release, it shows `Welcome to {title}`.

```
messages.po

#: src/app/pages/Program/components/ProgramContent.tsx:19
msgid "Welcome to {title}"
msgstr "Welcome to {title}"

```

Compiled messages returns: 

```
'Welcome to {title}': ['Welcome to ', ['title']],
```

**Comment by tricoder42** [comment]
It should be without `i18n._`:

```js
<WelcomeTitle>
   {t`Welcome to ${title}`}
</WelcomeTitle>
```

but ideally, you should use `Trans`:

```js
<WelcomeTitle>
   <Trans>Welcome to {title}</Trans>
</WelcomeTitle>
```

**Comment by tricoder42** [comment]
This should be fixed in latest release `3.0.0-20`
